modules/rotating-bound-service-keys.adoc

```diff
@@ -0,0 +1,408 @@
+[id="rotating-bound-service-keys_{context}"]
+ifdef::aws[= Rotating {aws-short} OIDC bound service account signer keys]
+ifdef::gcp[= Rotating {gcp-short} OIDC bound service account signer keys]
+ifdef::azure[= Rotating {azure-short} OIDC bound service account signer keys]
+
+If the Cloud Credential Operator (CCO) for your {product-title} cluster 
+ifdef::aws[on {aws-first}]
+ifdef::gcp[on {gcp-first}]
+ifdef::azure[on {azure-first}]
+is configured to operate in manual mode with 
+ifdef::aws[{sts-short},]
+ifdef::gcp[{gcp-wid-short},]
+ifdef::azure[{entra-short},]
+you can rotate the bound service account signer key.
+
+To rotate the key, you delete the existing key on your cluster, which causes the Kubernetes API server to create a new key.
+To reduce authentication failures during this process, you must immediately add the new public key to the existing issuer file. 
+Once the cluster is using the new key for authentication, you can remove any remaining keys.
+
+//Modified version of the disclaimer from enabling Azure WID on an existing cluster, since there are similar concerns:
+[IMPORTANT]
+====
+The process to rotate OIDC bound service account signer keys is disruptive and takes a significant amount of time.
+Some steps are time-sensitive.
+Before proceeding, observe the following considerations:
+
+* Read the following steps and ensure that you understand and accept the time requirement.
+The exact time requirement varies depending on the individual cluster, but it is likely to require at least one hour.
+
+* To reduce the risk of authentication failures, ensure that you understand and prepare for the time-sensitive steps.
+
+* During this process, you must refresh all service accounts and restart all pods on the cluster.
+These actions are disruptive to workloads.
+To mitigate this impact, you can temporarily halt these services and then redeploy them when the cluster is ready.
+====
+
+.Prerequisites
+
+* You have access to the {oc-first} as a user with the `cluster-admin` role.
+ifdef::aws[* You have access to the AWS CLI (`aws`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+ifdef::gcp[* You have access to the Google Cloud CLI (`gcloud`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+ifdef::azure[* You have access to the Azure CLI (`az`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+* You have configured the `ccoctl` utility.
+* Your cluster is in a stable state.
+You can confirm that the cluster is stable by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-stable-cluster --minimum-stable-period=5s
+----
+
+.Procedure
+
+. Configure the following environment variables:
++
+[source,text]
+----
+CLUSTER_ID=$(oc get infrastructures cluster -o jsonpath='{.status.infrastructureName}')
+CLUSTER_NAME=${CLUSTER_ID:0:-6}
```

Comment:
Update this to use the following (less opinionated) command. I've posted a [PR](https://github.com/openshift/cloud-credential-operator/pull/816) to the repo docs as well.
```CLUSTER_NAME=${CLUSTER_ID%-*}```

Reply:
I feel like I am very close to knowing how to formulate an `oc get` for that value, but not quite there. What would they need to send to get that `metadata.name` value?

Reply:
I _think_ it's:
```
$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
```

Reply:
oh wait that's the ting we just used :facepalm: